Snapshot results on iterating (#258)
realm/realm-dart: CHANGELOG.md

### Compatibility
* Dart ^2.15 on Windows, MacOS and Linux

### Fixed

> * Snapshot the results collection when Iterating if the items are realm objects. ([#258](https://github.com/realm/realm-dart/pull/258))

Reviewer: ```suggestion
* Snapshot the results collection when iterating if the items are realm objects. ([#258](https://github.com/realm/realm-dart/pull/258))
```

> * Iterating RealmResult will snapshot the collection if the items are RealmObject from Realm. ([#258](https://github.com/realm/realm-dart/pull/258))

Reviewer 2: ```suggestion
* Snapshot the results collection when Iterating if the items are realm objects. ([#258](https://github.com/realm/realm-dart/pull/258))
```
